WordPress/Classes/Utility/DiskCache.swift

```diff
@@ -163,8 +130,71 @@ public actor DiskCache {
 
     /// Returns the URL for the given cache key.
     public nonisolated func fileURL(for key: String) -> URL? {
-        guard let filename = key.sha1 else { return nil }
-        return rootURL.appendingPathComponent(filename, isDirectory: false)
+        sync.fileURL(for: key)
+    }
+
+    // MARK: SyncDiskCache
+
+    /// Returns the underlying disk cache that can be used synchronously.
+    ///
+    /// - warning: While disk cache techincally is thread-safe, it's not recommended
+    /// to use ``SyncDiskCache``. Make sure to use the async APIs.
+    public nonisolated var sync: SyncDiskCache { SyncDiskCache(rootURL: rootURL) }
+
+    public struct SyncDiskCache {
```

Comment:
It's not ideal that we have two different types that do the same thing. I'm guessing the stats somehow requires the cache to be read/written synchronously? Did you find out why there is a such requirement, and can it be refactored to work with async saves? If that's too hard, maybe we can keep the cache data in memory temporarily, until they are serialized? Probably would increase complexity in `DiskCache`, I'm not sure if that's a good idea...

Reply:
> I'm guessing the stats somehow requires the cache to be read/written synchronously?

I would rather not try and make it asynchronous because the queries/actions/events in the stats stores are out of control.

> It's not ideal that we have two different types that do the same thing.

I can't say that I'm happy about this code either. Alternatives I considered:

- Adding `nonisolated` synchronous versions of read/write APIs. It's a more standard approach, but I wanted to make the async methods stand out – they are the preferred way to use it.
- Making `DiskCache` a regular class and providing dedicated async functions for read/write. Perform the work on a serial DispatchQueue. It is a pretty standard and more conservative approach. But I wanted to give actors a try. They are a bit limiting, but this use case is simple enough that they can support it.
- Not using Swift Concurrency; making `DiskCache` fully synchronous. It is the simplest option. FileManager is fully thread-safe, so we are not really benefiting from actor isolation.

Reply:
> If that's too hard, maybe we can keep the cache data in memory temporarily, until they are serialized?

👀 the [code](https://github.com/kean/Nuke/blob/main/Sources/Nuke/Caching/DataCache.swift) that I based it on has a staging area for a similar reason; it's a bit of an overkill. It makes writes "instant", but reads still have to go to disk.

Reply:
I think I can get away with async writes in the case of the stores. They are fast enough (usually <5ms), but they still can and will lead to lost frames. They are also fast enough that "the chances are near zero" that the next read will be a miss. If is is a miss, it's not a big deal – we'll just fetch from the network. It's cache. It has no availability guarantees.